### PR TITLE
lxd: update to 3.18.

### PR DESCRIPTION
**Pull request by CameronNemo**

### CODE REVIEW

**Comment by CameronNemo** [comment]
build passes sans lint.

althoUGH, seeing this while testing.

```
Error: Failed instance creation: Create container from image: Unpack image: Unpack failed, Failed to run: unsquashfs -f -d /var/lib/lxd/storage-pools/default/containers/knowing-louse/rootfs -n /var/lib/lxd/images/f009e1cd8d9e1064a8db1e68fa400cd596db8e5fbfa2cbd3f62d28a202e7d5b0.rootfs: FATAL ERROR:write_file: failed to create file /var/lib/lxd/storage-pools/default/containers/knowing-louse/rootfs/usr/lib/udev/hwdb.d/20-bluetooth-vendor-product.hwdb, because Too many open files.
```

File: srcpkgs/dqlite/patches/musl.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git src/vfs.c src/vfs.c
+index 31e138c..5d2a7c4 100644
+--- src/vfs.c
++++ src/vfs.c
+@@ -1,6 +1,7 @@
+ #include <string.h>
+ #include <sys/time.h>
+ #include <errno.h>
++#include <pthread.h>
```
Comment (fosslinux) [inline srcpkgs/dqlite/patches/musl.patch]: While cross-building for musl I didn't see this needed.

Comment (CameronNemo) [reply srcpkgs/dqlite/patches/musl.patch]: I submitted it upstream. I think it was released and merged.

**Comment by CameronNemo** [comment]
@fosslinux @julio641742 mind testing this? I applied a workaround to an LXC bug in the most recent push. Now having success with launching and exec'ing into containers.

**Comment by fosslinux** [comment]
I can also confirm, it works so far for me. Will update with any further problems.

**Comment by fosslinux** [comment]
Looks like licenses need to be fixed in templates.

**Comment by CameronNemo** [comment]
Which licenses? Are you referencing the lint?

File: srcpkgs/libco/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+homepage="https://github.com/canonical/libco"
+distfiles="https://github.com/canonical/libco/archive/v${version}.tar.gz"
+checksum=6d6d6ada15a36224ee4aa59b8361c4b90f1ffb6c077b4932aa6e6344ee91be0e
+
+post_install() {
+	vlicense LICENSE
+}
+
+libco-devel_package() {
+	depends="libco>=${version}_${revision}"
```
Comment (abenson) [inline srcpkgs/libco/template]: ${sourcepkg}

Comment (CameronNemo) [reply srcpkgs/libco/template]: fixed x3

File: srcpkgs/dqlite/template [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
 revision=1
 build_style=gnu-configure
 hostmakedepends="pkg-config autoconf automake libtool"
-makedepends="libuv-devel sqlite-replication-devel"
+makedepends="sqlite-replication-devel raft-devel libco-devel"
 depends="sqlite-replication"
 short_desc="Distributed SQLite database using the Raft algorithm"
 maintainer="Cameron Nemo <camerontnorman@gmail.com>"
 license="Apache-2.0"
 homepage="https://github.com/CanonicalLtd/dqlite"
```
Comment (fosslinux) [inline srcpkgs/dqlite/template]: CanonicalLTD is now just canonical; seems like the two orgs have been merged.

File: srcpkgs/dqlite/template [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
+version=1.1.0
 revision=1
 build_style=gnu-configure
 hostmakedepends="pkg-config autoconf automake libtool"
-makedepends="libuv-devel sqlite-replication-devel"
+makedepends="sqlite-replication-devel raft-devel libco-devel"
 depends="sqlite-replication"
 short_desc="Distributed SQLite database using the Raft algorithm"
 maintainer="Cameron Nemo <camerontnorman@gmail.com>"
 license="Apache-2.0"
```
Comment (fosslinux) [inline srcpkgs/dqlite/template]: LGPLv3 https://github.com/canonical/dqlite/commit/fe0fbed8ca717947f90bf2018b28d06644d307b0

File: srcpkgs/libco/template [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'libco'
+pkgname=libco
+version=20
+revision=1
+build_style=gnu-makefile
+short_desc="Cooperative multithreading library written in C89"
+maintainer="Julio Galvan <juliogalvan@protonmail.com>"
+license="ISC"
+homepage="https://github.com/canonical/libco"
```
Comment (fosslinux) [inline srcpkgs/libco/template]: IMO, this should be the homepage https://code.byuu.org/libco here but keep the distfiles to make it far simpler.

File: srcpkgs/raft/template [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+version=0.9.9
+revision=1
+build_style=gnu-configure
+configure_args="--enable-example=no"
+hostmakedepends="pkg-config autoconf automake libtool"
+makedepends="libuv-devel"
+short_desc="C implementation of the Raft consensus protocol"
+maintainer="Julio Galvan <juliogalvan@protonmail.com>"
+license="Apache-2.0"
+homepage="https://github.com/CanonicalLtd/raft"
```
Comment (fosslinux) [inline srcpkgs/raft/template]: Same as above CanonicalLTD -> canonical.

File: srcpkgs/raft/template [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+pkgname=raft
+version=0.9.9
+revision=1
+build_style=gnu-configure
+configure_args="--enable-example=no"
+hostmakedepends="pkg-config autoconf automake libtool"
+makedepends="libuv-devel"
+short_desc="C implementation of the Raft consensus protocol"
+maintainer="Julio Galvan <juliogalvan@protonmail.com>"
+license="Apache-2.0"
```
Comment (fosslinux) [inline srcpkgs/raft/template]: Similarily also LGPLv3 https://github.com/canonical/raft/commit/e2d328fe309b14152f328b9d293f5343d842dcfe

File: common/shlibs [hunk 8]
```diff
@@ -3676,3 +3676,5 @@ libgcj-tools.so.17 libgcj-6.5.0_1
 libgcj_bc.so.1 libgcj-6.5.0_1
 libgcj.so.17 libgcj-6.5.0_1
 libgij.so.17 libgcj-6.5.0_1
+libco.so.0 libco-1.0.2_1
```
Comment (fosslinux) [inline common/shlibs]: Is libco not 20_1?

File: srcpkgs/libco/template (hunk 5 above)
Comment (CameronNemo) [reply srcpkgs/libco/template]: There are changes to the github repo that have not been upstreamed and will likely never be upstreamed. So I would rather keep the homepage as-is.

File: srcpkgs/dqlite/template (hunk 4 above)
Comment (CameronNemo) [reply srcpkgs/dqlite/template]: Could not find an SPDX for LGPL-3.0-only with linking exception, so used custom. Also xlint will still complain due to https://github.com/leahneukirchen/xtools/pull/146 .

**Comment by fosslinux** [comment]
Hey @CameronNemo I have been running this for my day-to-day stuff for around 2 months now, is this good to go?

**Comment by CameronNemo** [comment]
Rebased and pushed, waiting on CI. Once it is green you can try to ask someone with commit privileges to merge it.

**Comment by CameronNemo** [comment]
CI fluke. I built with x86_64 and it was fine.

**Comment by fosslinux** [comment]
Is this superseded by #18327?

**Comment by CameronNemo** [comment]
yeah

**Pull request closed without merging**